src/http/GraphClientFactory.ts

```diff
@@ -0,0 +1,56 @@
+import { BaseBearerTokenAuthenticationProvider } from "@microsoft/kiota-abstractions";
+import { Middleware } from "@microsoft/kiota-http-fetchlibrary";
+import { GraphHttpClient } from "./GraphHttpClient";
+import { getDefaultMiddlewares, GraphTelemetryOption } from "../middleware";
+
+/**
+ * Factory class for creating instances of `GraphHttpClient`.
+ */
+export class GraphClientFactory {
```

Comment:
we probably do not need the static class, a static function should be enough here